server/src/storage/segment.rs

```diff
@@ -128,14 +121,15 @@ impl UncompressedSegment for SegmentBuilder {
 
     /// Spill the in-memory segment to an Apache Parquet file and return Ok when finished.
     fn spill_to_apache_parquet(&mut self, key: String) -> Result<SpilledSegment, std::io::Error> {
-        let batch = self.get_record_batch();
+        let batch = self.get_record_batch().unwrap();
```

Comment:
Maybe either convert to `IOError` and return with `?` or document with a comment why `unwrap()` is ok.

Reply:
Added comment explaining why `unwrap` is safe to use here.